src/lib/homestore_backend/hs_homeobject.cpp

```diff
@@ -298,9 +296,22 @@ HSHomeObject::~HSHomeObject() {
     }
     trigger_timed_events();
 #endif
+
+    start_shutting_down();
+    // Wait for all pending requests to complete
+    while (true) {
+        auto pending_reqs = get_pending_request_num();
```

Comment:
baseline resync be considered as HS or HO?

Reply:
it will be considered as HS. 

Homeobject side graceful shutdown will only make sure that when shutting down homeobject ,  there is no pending external (or user api) calls from upper layer and reject later api calls .  

baseline resync is a mechanism of statemachin(raft), and all the baseline resync related code in homeobject is called by statemachine. homeobject can not do anything for this.  so , we need to handle this in raft level (homestore).

i will try to submit another pr for homestore graceful shutdown.